client/client.go

```diff
@@ -256,7 +258,9 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	}
 
 	if v != nil {
-		if err := json.Unmarshal(rsp.Result, v); err != nil {
+		dec := json.NewDecoder(strings.NewReader(string(rsp.Result)))
```

Comment:
the `string` here forces a spurious copy of `rsp.Result`. I think you should use a `bytes.NewReader` here instead.

You've got a lot of them; all instances of `strings.NewReader(string(thing)` where the `thing` is a `[]byte` (o a `json.RawMessage`, same thing) should be a `bytes.NewReader(thing)` directly. As far as I can tell that's all instances of `strings.Newreader(string(thing))` in this code.